ajax/timeline.php

```diff
@@ -116,6 +116,7 @@
         }
         $foreignKey = $parent->getForeignKeyField();
         $params[$foreignKey] = $_REQUEST[$foreignKey];
+        /** @var CommonITILObject $parent */
```

Comment:
If the `parenttype` parameter isn't a CommonITILObject class name, I think the script should exit just as if it wasn't specified at all. So, a `is_subclass_of` check earlier should remove the need for this PHPDoc.